core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -756,4 +756,54 @@ public void testUpdateSchema() {
     Assert.assertEquals("Should return expected last column id",
         6, threeSchemaTable.lastColumnId());
   }
+
+  @Test
+  public void testCreateV2MetadataThroughTableProperty() {
```

Comment:
Should we also add the unit tests in flink/spark/hive module to verify the end-to-end DDL work ?  If so, I think we could make it to be a separate PR for this.

Reply:
> It is a bit hard to test this in unit test because format version is not publicly accessible. I have tested with Spark on EMR and manually verified that the metadata file shows the correct version.

I think we could access the format version by using the following code: 

```java
Table table = ...
TableOperations ops = ((BaseTable) table).operations();
TableMetadata meta = ops.current();
int formatVersion = meta.formatVersion();
```